From 6e39d995b17ce6923d1c86548b58f400f762fb63 Mon Sep 17 00:00:00 2001
From: Yann Dirson <yann.dirson@vates.tech>
Date: Wed, 23 Jul 2025 19:36:10 +0200
Subject: [PATCH] Use EFI/almalinux

We don't want to rebuild our upstream packages just for a path change.
---
 Makefile                  |  2 +-
 backend.py                | 10 +++++-----
 bootloader/grub-usb.patch |  4 ++--
 upgrade.py                |  2 +-
 4 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/Makefile b/Makefile
index d507958..fe55516 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 # destinations
 DESTDIR =
 INSTALLER_DIR = /opt/xensource/installer
-EFI_DIR = /EFI/xenserver
+EFI_DIR = /EFI/almalinux
 SERVICE_DIR = usr/lib/systemd/system
 # multipath.conf to be taken as a base
 XS_MPATH_CONF = /etc/multipath.conf
diff --git a/backend.py b/backend.py
index f4c6856..39b116b 100644
--- a/backend.py
+++ b/backend.py
@@ -1086,7 +1086,7 @@ def installBootLoader(mounts, disk, boot_partnum, primary_partnum, branding,
     if host_config:
         s = serial and {'port': serial.id, 'baud': int(serial.baud)} or None
 
-        fn = os.path.join(mounts['boot'], "efi/EFI/xenserver/grub.cfg")
+        fn = os.path.join(mounts['boot'], "efi/EFI/almalinux/grub.cfg")
         boot_config = bootloader.Bootloader('grub2', fn,
                                             timeout=constants.BOOT_MENU_TIMEOUT,
                                             serial=s, location=location)
@@ -1131,10 +1131,10 @@ def setEfiBootEntry(mounts, disk, boot_partnum, install_type, branding):
                                  "Failed to remove efi boot entry %r" % (line,))
 
     # Then add a new one
-    if os.path.exists(os.path.join(mounts['esp'], 'EFI/xenserver/shimx64.efi')):
-        efi = "EFI/xenserver/shimx64.efi"
-    elif os.path.exists(os.path.join(mounts['esp'], 'EFI/xenserver/grubx64.efi')):
-        efi = "EFI/xenserver/grubx64.efi"
+    if os.path.exists(os.path.join(mounts['esp'], 'EFI/almalinux/shimx64.efi')):
+        efi = "EFI/almalinux/shimx64.efi"
+    elif os.path.exists(os.path.join(mounts['esp'], 'EFI/almalinux/grubx64.efi')):
+        efi = "EFI/almalinux/grubx64.efi"
     else:
         raise RuntimeError("Failed to find EFI loader")
     rc, err = util.runCmd2(["chroot", mounts['root'], "/usr/sbin/efibootmgr", "-c",
diff --git a/bootloader/grub-usb.patch b/bootloader/grub-usb.patch
index 84fec44..2c88a8a 100644
--- a/bootloader/grub-usb.patch
+++ b/bootloader/grub-usb.patch
@@ -1,5 +1,5 @@
---- /EFI/xenserver/grub.cfg	2024-03-18 13:30:12.000000000 +0100
-+++ /EFI/xenserver/grub-usb.cfg	2024-03-18 13:30:12.000000000 +0100
+--- /EFI/almalinux/grub.cfg	2024-03-18 13:30:12.000000000 +0100
++++ /EFI/almalinux/grub-usb.cfg	2024-03-18 13:30:12.000000000 +0100
 @@ -15,6 +15,7 @@
  insmod ext2
  
diff --git a/upgrade.py b/upgrade.py
index aad6ed5..46bf874 100644
--- a/upgrade.py
+++ b/upgrade.py
@@ -380,7 +380,7 @@ class ThirdGenUpgrader(Upgrader):
                     if os.path.exists(src):
                         util.runCmd2(['cp', '-f', src, os.path.join(backup_fs.mount_point, destination)])
 
-                configMaps = [("efi-grub.cfg", "boot/efi/EFI/xenserver/grub.cfg"),
+                configMaps = [("efi-grub.cfg", "boot/efi/EFI/almalinux/grub.cfg"),
                                 ("grub.cfg", "boot/grub"),
                                 ("menu.lst", "boot/grub"),
                                 ("extlinux.conf", "boot")]
-- 
2.39.5

